### ISSUE TITLE
Module graph not working in workspace

### ISSUE THREAD

**Issue by TECH7Fox** [vitest-dev/vitest#3470]
### Describe the bug

I have a workspace with multiple projects. When I run `vitest --ui --coverage` all the module graphs are empty. They do work if I run vitest in the project directly.

### Reproduction

https://github.com/TECH7Fox/vitest-monorepo

### System Info

```shell
System:
    OS: Linux 5.15 Ubuntu 20.04.6 LTS (Focal Fossa)
    CPU: (8) x64 11th Gen Intel(R) Core(TM) i5-1145G7 @ 2.60GHz
    Memory: 2.60 GB / 15.36 GB
    Container: Yes
    Shell: 5.0.17 - /bin/bash
  Binaries:
    Node: 18.14.0 - ~/.nvm/versions/node/v18.14.0/bin/node
    Yarn: 1.22.19 - /usr/local/bin/yarn
    npm: 9.6.7 - ~/repositories/my_project/node_modules/.bin/npm
  Browsers:
    Chrome: 113.0.5672.92
    Firefox: 113.0.2
  npmPackages:
    @vitest/coverage-c8: ^0.31.1 => 0.31.1 
    vitest: ^0.31.1 => 0.31.1
```

### Used Package Manager

npm

### Validations

- [X] Follow our [Code of Conduct](https://github.com/vitest-dev/vitest/blob/main/CODE_OF_CONDUCT.md)
- [X] Read the [Contributing Guidelines](https://github.com/vitest-dev/vitest/blob/main/CONTRIBUTING.md).
- [X] Read the [docs](https://vitest.dev/guide/).
- [X] Check that there isn't [already an issue](https://github.com/vitest-dev/vitest/issues) that reports the same bug to avoid creating a duplicate.
- [X] Check that this is a concrete bug. For Q&A open a [GitHub Discussion](https://github.com/vitest-dev/vitest/discussions) or join our [Discord Chat Server](https://chat.vitest.dev).
- [X] The provided reproduction is a [minimal reproducible example](https://stackoverflow.com/help/minimal-reproducible-example) of the bug.

**Comment by TECH7Fox** [comment]
Here is a reproduction: https://github.com/TECH7Fox/vitest-monorepo

**Comment by TECH7Fox** [comment]
Here is the module graph when I run vitest in the project itself.
![image](https://github.com/vitest-dev/vitest/assets/32220029/e3b25794-ba87-442e-82cb-d69d10682a14)

And here is the module graph when I run vitest from the workspace. Empty.
![image](https://github.com/vitest-dev/vitest/assets/32220029/6a1f4e5a-e150-4915-8fb9-36916a0cc73d)

**Comment by TECH7Fox** [comment]
@sheremet-va any idea what could cause this?

**Comment by sheremet-va** [comment]
> @sheremet-va any idea what could cause this?

The module graph is now stored inside each workspace, but UI was not updated to reflect that.

**Comment by sheremet-va** [comment]
This should be fixed in Vitest 2.0.
